### ISSUE TITLE
class schema

### ISSUE THREAD

**Issue by Momellouky** [Momellouky/adr_watermark#1]
I raised this issue to discuss about the class schema. Here we will decide the best architecture and the best schema of the class. 
The main aim of this class is to be reusable. 

First, we need to establish a class name, class atributes and methods.

**Comment by Momellouky** [comment]
Good question! 

The aim of the repo itself is to create a watermarking library that we wil use in our experiment. However, I hope to create a reusable piece of code. In other words, I aim to create a library that we will use for our experiments and for our product (that we will build at the end). 

Therefore, I don't know how many classes we will have. We will check it here. 

As the purpose is watermarking. 
We need a watermarking class. A class that is able to read an image, adds the watermark and export it. 
therefore, two main functionalities (until now): reading and exporting an image, and adding a watermark. 

Reading and writing to a file system should be a seperate class that Implements the strategy pattern. Why ?. 
Now we read from the filesystem, tomorrow we may read from another source. Using the strategy pattern makes it easy to extend the code with additional functionalities. 

Even the watermarking class may Implement a strategy pattern (I don't know what do you think). In fact, now we use the DCT, If we come up with a new method, it should be easy to add the new method without modifing the existing code. 

To sum up, I guess we need three classes
- reader : reads an image (from a filesystem until now) 
- writer : writes the watermarked image 
- watermaker : the class that adds the watermark 

What do you think ?
